agent/stats/task_windows.go

```diff
@@ -0,0 +1,179 @@
+	"time"
+
+	"github.com/aws/amazon-ecs-agent/agent/api/task"
+	"github.com/aws/amazon-ecs-agent/agent/stats/resolver"
+
+	dockerstats "github.com/docker/docker/api/types"
+	"github.com/pkg/errors"
+)
+
+// Making it visible for unit testing
+var execCommand = exec.Command
+
+type StatsTask struct {
+	*statsTaskCommon
+}
+
+func newStatsTaskContainer(taskARN string, containerPID string, numberOfContainers int,
+	resolver resolver.ContainerMetadataResolver, publishInterval time.Duration, taskENIs task.TaskENIs) (*StatsTask, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+
+	devices := make([]string, len(taskENIs))
+	for index, device := range taskENIs {
+		devices[index] = device.LinkName
+	}
+
+	return &StatsTask{
+		statsTaskCommon: &statsTaskCommon{
+			TaskMetadata: &TaskMetadata{
+				TaskArn:          taskARN,
+				ContainerPID:     containerPID,
+				DeviceName:       devices,
+				NumberContainers: numberOfContainers,
+			},
+			Ctx:                   ctx,
+			Cancel:                cancel,
+			Resolver:              resolver,
+			metricPublishInterval: publishInterval,
+		},
+	}, nil
+}
+
+func (taskStat *StatsTask) retrieveNetworkStatistics() (map[string]dockerstats.NetworkStats, error) {
+	if len(taskStat.TaskMetadata.DeviceName) == 0 {
+		return nil, errors.Errorf("unable to find any device name associated with the task %s", taskStat.TaskMetadata.TaskArn)
+	}
+
+	networkStats := make(map[string]dockerstats.NetworkStats, len(taskStat.TaskMetadata.DeviceName))
+	for _, device := range taskStat.TaskMetadata.DeviceName {
+		networkAdaptorStatistics, err := taskStat.getNetworkAdaptorStatistics(device)
+		if err != nil {
+			return nil, err
+		}
+		networkStats[device] = *networkAdaptorStatistics
+	}
+
+	return networkStats, nil
+}
+
+// getNetworkAdaptorStatistics returns the network statistics per container for the given network interface.
+func (taskStat *StatsTask) getNetworkAdaptorStatistics(device string) (*dockerstats.NetworkStats, error) {
+	// Ref: https://docs.microsoft.com/en-us/powershell/module/netadapter/get-netadapterstatistics?view=windowsserver2019-ps
+	// The Get-NetAdapterStatistics cmdlet gets networking statistics from a network adapter.
+	// The statistics include broadcast, multicast, discards, and errors.
+	cmd := "Get-NetAdapterStatistics -Name \"" + device + "\" | Format-List -Property *"
+	out, err := execCommand("powershell", "-Command", cmd).CombinedOutput()
+
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to run Get-NetAdapterStatistics for %s", device)
+	}
+	str := string(out)
+
+	lines := strings.Split(str, "\n")
+	m := make(map[string]string)
+	for _, line := range lines {
+		// populate all the network metrics in a map
+		kv := strings.Split(line, ":")
+		if len(kv) != 2 {
+			continue
+		}
+		key := strings.TrimSpace(kv[0])
+		value := strings.TrimSpace(kv[1])
+		m[key] = value
+	}
+
+	receivedBroadcastPackets, err := strconv.ParseUint(m["ReceivedBroadcastPackets"], 10, 64)
```

Comment:
can these keys (e.e. `ReceivedBroadcastPackets`) be trusted not to change in the future? Is there a more reliable way than hard-coding?

Reply:
Based on an earlier discussion with Justin, it seems like this would be the way to go for now. 

We can possibly add an integ test for this which would execute this method and check that the stats are available. That way we can ensure that all the keys are as it is with each update.
Thoughts?

Reply:
This is a public API via Microsoft WMI. If they broke this it would affect more than just us. While it seems possible I think its safe to assume there would not be a breaking change here intentionally

Reply:
@rawahars - The more I think about it there is no real reason not to warn on this anyways. Since they are all `ParseUint` we could add a func like this:

```
func getMapValue(m map[string]string, key string) (uint, err) {
  v, ok := m[key]
  if !ok {
    return 0, errors.Errorf("failed to find key: %s in output", key)
  }
  val, err := strconv.ParseUint(v, 10, 64)
  if err != nil
    return 0, errors.Errorf("failed to parse network stats for %s with value: %s", key, v)
  }
  return val, nil
}
```

And then on the caller we use
```
recievedBrodcastPackets, err := getMapValue(m, "ReceivedBroadcastPackets")
if err != nil {
  return nil, err
}
// ...
```

And we handle both cases of both noticing that a key is not there which is unexpected as well as being more generic about the handling logic for all keys. What do you think?

Reply:
This seems like a good idea. I will go ahead and make the changes!